Mathlib/Analysis/Meromorphic/Divisor.lean

```diff
@@ -167,6 +186,62 @@ theorem divisor_inv {f : 𝕜 → 𝕜} :
   · simp [divisor_apply, h, meromorphicOrderAt_inv]
   · simp [divisor_def, h]
 
+/-- The divisor of the inverse is the negative of the divisor. -/
+@[simp]
+theorem divisor_fun_inv {f : 𝕜 → 𝕜} : divisor (fun z ↦ (f z)⁻¹) U = -divisor f U := divisor_inv
+
+open Classical in
+/--
+If `f` is meromorphic, then the divisor of `f ^ n` is `n` times the divisor of `f`.
+-/
+theorem divisor_pow {f : 𝕜 → 𝕜} (hf : MeromorphicOn f U) (n : ℕ) :
+    divisor (f ^ n) U = n • divisor f U := by
```

Comment:
Can you maybe also prove
```
theorem divisor_pow' (f : 𝕜 → 𝕜) {n : ℕ} (hn : n ≠ 0) :
    divisor (f ^ n) U = n • divisor f U := by
```
?

Reply:
@YaelDillies Thanks for looking at this! I believe the statement `divisor_pow'` is not correct. Take a meromorphic function and modify it by randomly changing the sign. Then the function is no longer meromorphic, but its square is… Am I making a mistake here?

Reply:
Oh, yes of course 🤦